### ISSUE TITLE
astropy4.0 precision loss reason found.

### ISSUE THREAD

**Issue by luojing1211** [nanograv/PINT#609]
I tracked down the reason for the precision loss under astropy4.0. It is actually due to astropy4.0 table somehow does not keep the location in the Time object when a time object put into a table. 
Here is a test code:
```
>>> import astropy
>>> from astropy.time import Time
>>> import numpy as np
>>> import astropy.units as u
>>> from astropy.table import Table

>>> version = astropy.__version__
>>> t2 = Time(58000.012345678, location=(2390490., -5564764., 1994727.)* u.m, scale='utc', format='mjd', precision=9)
>>> t2_list = [t2,]
>>> t2_table = Table([t2_list,], names=("mjd",))
>>> print("astropy table version: ", version, t2_table['mjd'][0].location, "\nlist version: ", t2_list[0].location)
```
Output in astropy3
```
astropy table version:  3.2.1 (2390490., -5564764., 1994727.) m 
list version:  (2390490., -5564764., 1994727.) m
```

Output in astropy4
```
astropy table version:  4.0 None 
list version:  (2390490., -5564764., 1994727.) m
```

@mhvk Should I post this to the astropy github issues?

**Comment by paulray** [comment]
Nice detective work!

**Comment by mhvk** [comment]
That is nice sleuthing indeed!  Yes, this should be raised as an issue, though in the meantime it may be possible to work around it by not making `Time` a list in the first place, but rather use an array of time. E.g.,
```
t3 = t2.reshape(1)
t3
# <Time object: scale='utc' format='mjd' value=[58000.01234568]>
t3_table = Table([t3], names=["mjd"])
t3_table["mjd"][0].location                                                                    
# <EarthLocation (2390490., -5564764., 1994727.) m>
```

p.s. In more detail, the trouble may be that the table initialization has changed to use `Time` itself to initialize from a list of times, and that this does not keep location information:
```
print(Time(t2).location)
# None
```
It seems this latter behaviour has been present since at least 3.1. Upstream bug report at https://github.com/astropy/astropy/issues/9962

**Comment by mhvk** [comment]
p.p.s. My suggestion of using array-shaped times should be backwards compatible, so hopefully that can get PINT working with every supported version of astropy again.

**Comment by mhvk** [comment]
p.p.p.s. Maybe an idea to have a cron job with astropy-dev - breakage of PINT would definitely be a reason to revert/fix whatever commit causes it! This can be run on travis - see astropy's `.travis.yml` for inspiration. Note that one can also use travis for big-endian architectures (s390).

**Comment by luojing1211** [comment]
Solved by PR #612
